licenses/_headers/license-prepend-missing.sh

```diff
@@ -29,15 +32,60 @@ find $SOURCE_DIR -type f "${ALWAYS_IGNORE[@]}" \
 echo "--------------------------------------------------------------------------------"
 echo "    BEGIN PREPENDING:"
 echo "--------------------------------------------------------------------------------"
+
+
+############################################################
+# Prepend license header to python files with a shebang and an encoding header.
+# Will ignore files with a current license header.
+############################################################
+GET=( -iname "*.py" )
+
+find $SOURCE_DIR -type f \( "${GET[@]}" \) "${ALWAYS_IGNORE[@]}" "${PYTHON_IGNORE[@]}" \
+    | xargs -r grep -l --max-count=1 "#!" \
+    | xargs -r grep -l --max-count=1 "# -\*-" \
+    | xargs -r grep -L "$CURRENT" \
+    | while read SRC
+do
+    BN=`basename ${SRC}`
+    echo HEADING ${SRC}
+    cat ${SRC} | head -n 2 > /tmp/licHead
+    cat header-py >> /tmp/licHead
+    cat ${SRC} | tail -n +3 >> /tmp/licHead
+    chmod --reference=${SRC} /tmp/licHead
+    mv /tmp/licHead ${SRC}
+done
+
+
+#######################################################################
+# Prepend license header to python files with encoding header.
+# Will ignore files with a current license header.
+#######################################################################
+GET=( -iname "*.py" )
+
+find $SOURCE_DIR -type f \( "${GET[@]}" \) "${ALWAYS_IGNORE[@]}" "${PYTHON_IGNORE[@]}" \
+    | xargs -r grep -l --max-count=1 "# -\*-" \
+    | xargs -r grep -L "$CURRENT" \
+    | while read SRC
+do
+    BN=`basename ${SRC}`
+    echo HEADING ${SRC}
+    cat ${SRC} | head -n 1 > /tmp/licHead
+    cat header-py >> /tmp/licHead
+    cat ${SRC} | tail -n +2 >> /tmp/licHead
+    chmod --reference=${SRC} /tmp/licHead
+    mv /tmp/licHead ${SRC}
+done
+
+
 ############################################################
 # Prepend license header to python files without a shebang.
 # Will ignore files with a current license header.
 ############################################################
 GET=( -iname "*.py" )
 
 find $SOURCE_DIR -type f \( "${GET[@]}" \) "${ALWAYS_IGNORE[@]}" "${PYTHON_IGNORE[@]}" \
-    | xargs grep -L "#!" \
-    | xargs grep -L "$CURRENT" \
+    | xargs -r grep -L "#!" \
```

Comment:
It is safest to have a `-r` in each `xargs` call.  If the file list passed to `xargs` is empty, `xargs` returns `(standard input)`, which then causes error in the subsequent code.  With `-r`, `xargs` doesn't run and the result remains an empty string.